hi-ml-cpath/src/health_cpath/utils/tiles_selection_utils.py

```diff
@@ -120,6 +120,10 @@ def init_report_cases(self) -> Dict[str, List[str]]:
         report_cases.update({f"FN_{class_id}": [] for class_id in range(1, self.n_classes)})
         return report_cases
 
+    def _clear_cached_slides_heaps(self) -> None:
```

Comment:
Can I suggest to apply DRY? there are now 4 copies of the same code snippet, in lines 107, 108, 124, 125. How we initialize and how we clear can hence run out of sync.